File: Prototype/Html5RenderEngineTests.Web/Scripts/Framework/Control.js
/// <reference path="FrameworkElement.js" />

Control.BackgroundProperty = DependencyProperty.Register("Background", Control);
Control.prototype.GetBackground = function () {
    return this.GetValue(Control.BackgroundProperty);
};
Control.prototype.SetBackground = function (value) {
    this.SetValue(Control.BackgroundProperty, value);
};

Control.BorderBrushProperty = DependencyProperty.Register("BorderBrush", Control);
Control.prototype.GetBorderBrush = function () {
    return this.GetValue(Control.BorderBrushProperty);
};
Control.prototype.SetBorderBrush = function (value) {
    this.SetValue(Control.BorderBrushProperty, value);
};

Control.BorderThicknessProperty = DependencyProperty.Register("BorderThickness", Control);
Control.prototype.GetBorderThickness = function () {
    return this.GetValue(Control.BorderThicknessProperty);
};
Control.prototype.SetBorderThickness = function (value) {
    this.SetValue(Control.BorderThicknessProperty, value);
};

Control.FontFamilyProperty = DependencyProperty.Register("FontFamily", Control);
Control.prototype.GetFontFamily = function () {
    return this.GetValue(Control.FontFamilyProperty);
};
Control.prototype.SetFontFamily = function (value) {
    this.SetValue(Control.FontFamilyProperty, value);
};

Control.FontSizeProperty = DependencyProperty.Register("FontSize", Control);
Control.prototype.GetFontSize = function () {
    return this.GetValue(Control.FontSizeProperty);
};
Control.prototype.SetFontSize = function (value) {
    this.SetValue(Control.FontSizeProperty, value);
};

Control.FontStretchProperty = DependencyProperty.Register("FontStretch", Control);
Control.prototype.GetFontStretch = function () {
    return this.GetValue(Control.FontStretchProperty);
};
Control.prototype.SetFontStretch = function (value) {
    this.SetValue(Control.FontStretchProperty, value);
};

Control.FontStyleProperty = DependencyProperty.Register("FontStyle", Control);
Control.prototype.GetFontStyle = function () {
    return this.GetValue(Control.FontStyleProperty);
};
Control.prototype.SetFontStyle = function (value) {
    this.SetValue(Control.FontStyleProperty, value);
};

Control.FontWeightProperty = DependencyProperty.Register("FontWeight", Control);
Control.prototype.GetFontWeight = function () {
    return this.GetValue(Control.FontWeightProperty);
};
Control.prototype.SetFontWeight = function (value) {
    this.SetValue(Control.FontWeightProperty, value);
};

Control.ForegroundProperty = DependencyProperty.Register("Foreground", Control);
Control.prototype.GetForeground = function () {
    return this.GetValue(Control.ForegroundProperty);
};
Control.prototype.SetForeground = function (value) {
    this.SetValue(Control.ForegroundProperty, value);
};

Control.HorizontalContentAlignmentProperty = DependencyProperty.Register("HorizontalContentAlignment", Control);
Control.prototype.GetHorizontalContentAlignment = function () {
    return this.GetValue(Control.HorizontalContentAlignmentProperty);
};
Control.prototype.SetHorizontalContentAlignment = function (value) {
    this.SetValue(Control.HorizontalContentAlignmentProperty, value);
};

Control.IsEnabledProperty = DependencyProperty.Register("IsEnabled", Control, true);
Control.prototype.GetIsEnabled = function () {
    return this.GetValue(Control.IsEnabledProperty);
};
Control.prototype.SetIsEnabled = function (value) {
    this.SetValue(Control.IsEnabledProperty, value);
};

//Control.IsTemplateItemProperty;
//Control.IsTabStopProperty;
//Control.PaddingProperty;
//Control.TabIndexProperty;
//Control.TabNavigationProperty;
//Control.TemplateProperty;
//Control.VerticalContentAlignmentProperty;
//Control.DefaultStyleKeyProperty;
Control.prototype = new FrameworkElement;
Control.prototype.constructor = Control;
function Control() {
    FrameworkElement.call(this);
}

File: Source/WickedSick.Server.XamlParser/Elements/Controls/Control.cs
using WickedSick.Server.XamlParser.Elements.Core;
using WickedSick.Server.XamlParser.Elements.Input;
using WickedSick.Server.XamlParser.Elements.Media;
using WickedSick.Server.XamlParser.Elements.Types;

namespace WickedSick.Server.XamlParser.Elements.Controls
{
    [Element("Fayde.Controls")]
    public abstract class Control : FrameworkElement
    {
        public static readonly PropertyDescription BackgroundProperty = PropertyDescription.Register("Background", typeof(Brush), typeof(Control));
        public static readonly PropertyDescription BorderBrushProperty = PropertyDescription.Register("BorderBrush", typeof(Brush), typeof(Control));
        public static readonly PropertyDescription BorderThicknessProperty = PropertyDescription.Register("BorderThickness", typeof(Thickness), typeof(Control));
        public static readonly PropertyDescription FontFamilyProperty = PropertyDescription.Register("FontFamily", typeof(string), typeof(Control));
        public static readonly PropertyDescription FontSizeProperty = PropertyDescription.Register("FontSize", typeof(double), typeof(Control));
        public static readonly PropertyDescription FontStretchProperty = PropertyDescription.Register("FontStretch", typeof(FontStretch), typeof(Control));
        public static readonly PropertyDescription FontStyleProperty = PropertyDescription.Register("FontStyle", typeof(FontStyle), typeof(Control));
        public static readonly PropertyDescription FontWeightProperty = PropertyDescription.Register("FontWeight", typeof(FontWeight), typeof(Control));
        public static readonly PropertyDescription ForegroundProperty = PropertyDescription.Register("Foreground", typeof(Brush), typeof(Control));
        public static readonly PropertyDescription HorizontalContentAlignmentProperty = PropertyDescription.Register("HorizontalContentAlignment", typeof(HorizontalAlignment), typeof(Control));
        public static readonly PropertyDescription IsEnabledProperty = PropertyDescription.Register("IsEnabled", typeof(bool), typeof(Control));
        public static readonly PropertyDescription IsTabStopProperty = PropertyDescription.Register("IsTabStop", typeof(bool), typeof(Control));
        public static readonly PropertyDescription PaddingProperty = PropertyDescription.Register("Padding", typeof(Thickness), typeof(Control));
        public static readonly PropertyDescription TabIndexProperty = PropertyDescription.Register("TabIndex", typeof(int), typeof(Control));
        public static readonly PropertyDescription TabNavigationProperty = PropertyDescription.Register("TabNavigation", typeof(KeyboardNavigationMode), typeof(Control));
        public static readonly PropertyDescription TemplateProperty = PropertyDescription.Register("Template", typeof(ControlTemplate), typeof(Control));
        public static readonly PropertyDescription VerticalContentAlignmentProperty = PropertyDescription.Register("VerticalContentAlignment", typeof(VerticalAlignment), typeof(Control));
    }
}